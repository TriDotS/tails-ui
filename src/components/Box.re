let component = ReasonReact.statelessComponent("Box");

let make =
    (
      ~sm: string,
      ~md: string,
      ~lg: string,
      ~xl: string,
      ~children: array(ReasonReact.reactElement)
    ) => {
  ...component,
  render: _self =>
    <div className={j|w-$(sm) sm:w-$(sm) md:w-$(md) lg:w-$(lg) xl:w-$(xl)|j}>
      (ReasonReact.arrayToElement(children))
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~sm=jsProps##sm,
      ~md=jsProps##md,
      ~lg=jsProps##lg,
      ~xl=jsProps##xl,
      ~children=jsProps##children
    )
  );
