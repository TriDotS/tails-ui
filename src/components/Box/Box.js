import PropTypes from 'prop-types'
import React from 'react'

const Box = ({ sm, md, lg, xl, children }) => (
  <div className={`w-${sm} sm:w-${sm} md:w-${md} lg:w-${lg} xl:w-${xl}`}>
    {children}
  </div>
)

Box.defaultProps = {
  sm: 'full',
  padding: '0',
  margin: '0'
}

Box.propTypes = {
  sm: PropTypes.oneOf([
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '8',
    '10',
    '12',
    '16',
    '24',
    '32',
    '48',
    '64',
    'auto',
    'px',
    '1/2',
    '1/3',
    '1/4',
    '3/4',
    '1/5',
    '2/5',
    '3/5',
    '4/5',
    '1/6',
    '5/6',
    'full',
    'screen'
  ]),
  md: PropTypes.oneOf([
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '8',
    '10',
    '12',
    '16',
    '24',
    '32',
    '48',
    '64',
    'auto',
    'px',
    '1/2',
    '1/3',
    '1/4',
    '3/4',
    '1/5',
    '2/5',
    '3/5',
    '4/5',
    '1/6',
    '5/6',
    'full',
    'screen'
  ]),
  lg: PropTypes.oneOf([
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '8',
    '10',
    '12',
    '16',
    '24',
    '32',
    '48',
    '64',
    'auto',
    'px',
    '1/2',
    '1/3',
    '1/4',
    '3/4',
    '1/5',
    '2/5',
    '3/5',
    '4/5',
    '1/6',
    '5/6',
    'full',
    'screen'
  ]),
  xl: PropTypes.oneOf([
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '8',
    '10',
    '12',
    '16',
    '24',
    '32',
    '48',
    '64',
    'auto',
    'px',
    '1/2',
    '1/3',
    '1/4',
    '3/4',
    '1/5',
    '2/5',
    '3/5',
    '4/5',
    '1/6',
    '5/6',
    'full',
    'screen'
  ]),
  children: PropTypes.node.isRequired
}

export default Box
