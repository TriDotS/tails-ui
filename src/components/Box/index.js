import PropTypes from 'prop-types'
import React from 'react'

const Box = ({ sm, md, lg, xl, padding, margin, children }) => (
  <div
    className={`w-full sm:w-${sm} md:w-${md} lg:w-${lg} xl:w-${xl} p-${padding} m-${margin}`}
  >
    {children}
  </div>
)

Box.defaultProps = {
  sm: 'full',
  padding: '0',
  margin: '0'
}

Box.propTypes = {
  sm: PropTypes.string,
  md: PropTypes.string,
  lg: PropTypes.string,
  xl: PropTypes.string,
  padding: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  margin: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  children: PropTypes.node.isRequired
}

export default Box
