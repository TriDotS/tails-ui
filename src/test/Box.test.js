import Box from '../components/Box/Box'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Box>This is a box</Box>, div)
})

test('box snapshot', () => {
  const tree = renderer.create(<Box>This is a box</Box>).toJSON()
  expect(tree).toMatchSnapshot()
})
