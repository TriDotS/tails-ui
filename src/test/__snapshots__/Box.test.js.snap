// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`box snapshot 1`] = `
<div
  className="w-full sm:w-full md:w-undefined lg:w-undefined xl:w-undefined"
>
  This is a box
</div>
`;
